### ISSUE TITLE
[QUESTION] How does "sortBy": "most-used" work ?

### ISSUE THREAD

**Issue by Zixim** [Lissy93/dashy#1033]
### Question

I've set `"sortBy": "most-used",` for all of the sections on the dash, but none of the items inside of the sections are changing positions.
Does something else need to be set for sorting by most-used ?

### Category

Configuration

### Please tick the boxes

- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy (check the first two digits of the version number)
- [X] You've checked that this [question hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by hypatia1** [comment]
I also have this problem.

**Comment by Zixim** [comment]
This is still relevant.

**Comment by Lissy93** [comment]
Heya @Zixim 
Sorry for the delay...

So when `section[n].displayData.sortBy: most-used` is enabled for a given section, each time you click an item, a click count will be stored in local storage (using key `MOST_USED`). Then items are sorted based on this value.

Here's the code for the sorting:

https://github.com/Lissy93/dashy/blob/0319b8b1ed260ec04b36993a92930a5e723be397/src/components/LinkItems/Section.vue#L222-L228

And here's the code for the setting:

https://github.com/Lissy93/dashy/blob/0319b8b1ed260ec04b36993a92930a5e723be397/src/mixins/ItemMixin.js#L215-L222


Since data is only stored client-side, so if you open Dashy on a new device or browser, then usage counters won't be carried over, hence the order may vary between sessions.

**Comment by Zixim** [comment]
@Lissy93 thanks for your clarification. It just doesn't seem to be working at all.

Here's what is stored in Local Storage : 
![image](https://user-images.githubusercontent.com/7814664/221366215-bd20455d-a88d-477a-ba6a-eec9a2d779af.png)
This for a Dashy instance that has been set to sortby most-used for all sections for many weeks, i.e. : 
```sections:
  - name: Online media
    icon: fas fa-book
    displayData:
      sortBy: most-used
      rows: 1
      cols: 3
      collapsed: true
      hideForGuests: false
    items:
      .....
```

Something must be wrong somewhere ?

**Comment by Lissy93** [comment]
Oh yeah, you're right - looks like the `itemId` is `undefined` when it gets passed to that function. It should have been `this.item.id` instead of `this.id`. I've opened a PR to fix this in #1123 :)
